### PR TITLE
Add sonarcloud analysis

### LINKED ISSUE #569: Integrate SonarCloud scanning
**Describe the solution you'd like**
Perform integration of SonarCloud with the k8ssandra GH organization projects.

An initial SonarCloud organization has been created here to be used:  https://sonarcloud.io/organizations/k8ssandra/projects

Explore what is possible when integrating SC with GH.  Ideally, if possible, we'd like to see public reporting overall for the projects within SC's interface as well as automated scanning of all PRs before they are approved through GH Actions.

### PR DESCRIPTION
**Pull request by adejanovski**
**What this PR does**:
Enable Sonarcloud static analysis when commits are pushed to `main`: https://sonarcloud.io/dashboard?id=k8ssandra_k8ssandra

We cannot enable this on PRs as it requires the SONAR_TOKEN which is stored in a secret, and secrets aren't available to external PRs.

**Which issue(s) this PR fixes**:
Fixes #569

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/k8ssandra/templates/stargate/stargate.yaml [hunk 1]
```diff
@@ -111,11 +111,13 @@ spec:
               path: /checker/liveness
               port: health
             timeoutSeconds: 10
-            initialDelaySeconds: 240
+            failureThreshold: 10
```
Comment (burmanm) [inline charts/k8ssandra/templates/stargate/stargate.yaml:114]: Why are these modified?

Comment (adejanovski) [reply charts/k8ssandra/templates/stargate/stargate.yaml:114]: yup, my bad, it's a leftover from some other work I was doing.
Good catch 👍

Comment (adejanovski) [reply charts/k8ssandra/templates/stargate/stargate.yaml]: Removed 👌

**Review by burmanm — APPROVED** [review]
